sidecar/supervisor/appsupervisor.go

```diff
@@ -93,29 +118,42 @@ func (a *AppSupervisor) DoAppSupervision(agent *register.RegistrationAgent) {
 					log.Errorf("App failed to start: %v", err)
 				}
 			}
-
+			exit(a.Helpers, syscall.SIGKILL)
```

Comment:
It is better to try and gracefully terminate (SIGTERM) the helpers first before killing them.